superset-frontend/src/components/ErrorMessage/TimeoutErrorMessage.tsx

```diff
@@ -0,0 +1,234 @@
+  }
+
+  .icon {
+    margin-right: ${({ theme }) => 2 * theme.gridUnit}px;
+  }
+
+  .link {
+    color: ${({ theme }) => theme.colors.error.dark2};
+    text-decoration: underline;
+  }
+`;
+
+const ErrorModal = styled(Modal)`
+  color: ${({ theme }) => theme.colors.error.dark2};
+
+  .icon {
+    margin-right: ${({ theme }) => 2 * theme.gridUnit}px;
+  }
+
+  .header {
+    align-items: center;
+    background-color: ${({ theme }) => theme.colors.error.light2};
+    display: flex;
+    justify-content: space-between;
+    font-size: ${({ theme }) => theme.typography.sizes.l}px;
+
+    // Remove clearfix hack as Superset is only used on modern browsers
+    ::before,
+    ::after {
+      content: unset;
+    }
+  }
+`;
+
+const LeftSideContent = styled.div`
+  align-items: center;
+  display: flex;
+`;
+
+interface TimeoutErrorExtra {
+  location: 'chart' | 'dashboard' | 'sqllab';
+  owners?: string[];
+  timeout: number;
+}
+
+function TimeoutErrorMessage({
+  error,
+}: {
+  error: SupersetError<TimeoutErrorExtra>;
+}) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isMessageExpanded, setIsMessageExpanded] = useState(false);
+  const { extra } = error;
+
+  const title = ['chart', 'dashboard'].includes(extra.location)
+    ? tn(
+        'We’re having trouble loading this visualization. Queries are set to timeout after %s second.',
+        'We’re having trouble loading this visualization. Queries are set to timeout after %s seconds.',
+        extra.timeout,
+        extra.timeout,
+      )
+    : tn(
+        'We’re having trouble loading these results. Queries are set to timeout after %s second.',
+        'We’re having trouble loading these results. Queries are set to timeout after %s seconds.',
+        extra.timeout,
+        extra.timeout,
+      );
+
+  const message = (
+    <>
+      <p>
+        {t('This may be triggered by:')}
+        <br />
+        {ERROR_DATASOURCE_TOO_LARGE.message}{' '}
+        <a
+          href={ERROR_DATASOURCE_TOO_LARGE.link}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          <i className="fa fa-external-link" />
+        </a>
+        <br />
+        {ERROR_DATASOURCE_UNDER_LOAD.message}{' '}
+        <a
+          href={ERROR_DATASOURCE_UNDER_LOAD.link}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          <i className="fa fa-external-link" />
+        </a>
+      </p>
+      {['chart', 'dashboard'].includes(extra.location) && extra.owners && (
+        <>
+          <br />
+          <p>{t('Please reach out to the Chart Owner(s) for assistance.')}</p>
+          <p>{t('Chart Owner(s): %s', extra.owners.join(', '))}</p>
+        </>
+      )}
+    </>
+  );
+
+  const copyText = `${title}
+${t('This may be triggered by:')}
+${ERROR_DATASOURCE_TOO_LARGE.message}
+${ERROR_DATASOURCE_UNDER_LOAD.message}`;
+
+  return (
+    <ErrorAlert>
+      <div className="topRow">
+        <LeftSideContent>
```

Comment:
It was originally, but we needed the same styling in both the error alert and the modal, and the classes didn't work because the Modal is rendered outside the usual dom flow. The two choices were define the same class twice in 2 different styled components or create a new styled component just for this. I chose the latter, but can move back to the other way if you prefer

Reply:
Or maybe create a more specific component?

```ts
const ErrorTitleStyles = styled.div`
  display: flex;
  align-items: center;
`

function ErrorTitle({ children }: { children: ReactNode }) {
  return (
   <ErrorTitleStyles>
     <Icon
        className="icon"
        name="error"
        color={supersetTheme.colors.error.base}
     />
     {children || }
   </ErrorTitleStyles>
  }
)

<ErrorTitle>
  <strong>{t('Timeout Error')}</strong>
</ErrorTitle>

<ErrorTitle>
  <div className="title">{t('Timeout Error')}</div>
</ErrorTitle>
```